cmd/bucket-handlers.go

```diff
@@ -33,7 +32,7 @@ import (
 
 // http://docs.aws.amazon.com/AmazonS3/latest/dev/using-with-s3-actions.html
 // Enforces bucket policies for a bucket for a given tatusaction.
-func enforceBucketPolicy(bucket string, action string, reqURL *url.URL) (s3Error APIErrorCode) {
+func enforceBucketPolicy(bucket string, action string, req http.Request) (s3Error APIErrorCode) {
```

Comment:
I think it's generally a better practice to pass only the values that a function needs, in this case req.URL and referrer and not pass the request as a whole which contains other things not needed by this function. Function signature becomes self documenting - in the sense it tells us exactly what it depends on.

But if the function depends on most of the things contained in http.Request then it would be better to pass it entirely instead of making the argument very long compromising on code readability.